src/monitoring_tests/fee_test.py

```diff
@@ -0,0 +1,160 @@
+
+        if len(partially_fillable_indices) > 0:
+            # get additional data for the batch
+            receipt = self.web3_api.get_receipt(tx_hash)
+            if receipt is None:
+                return False
+
+            # get batch costs
+            gas_amount, gas_price = self.web3_api.get_batch_gas_costs(
+                transaction, receipt
+            )
+
+            for i in partially_fillable_indices:
+                trade = trades[i]
+                quote = self.orderbook_api.get_quote(trade)
+                if quote is None:
+                    self.logger.error("Error fetching quote. Skipping hash %s", tx_hash)
+                    return False
+
+                gas_price_quote = self.web3_api.get_current_gas_price()
+                if gas_price_quote is None:
+                    self.logger.error(
+                        "Error fetching current gas price. Skipping hash %s", tx_hash
+                    )
+                    return False
+
+                quote.adapt_execution_to_gas_price(gas_price_quote, gas_price)
+
+                fee_amount = trade.execution.fee_amount
+                quote_fee_amount = quote.execution.fee_amount
+
+                diff_fee_abs = fee_amount - quote_fee_amount
+                diff_fee_rel = (fee_amount - quote_fee_amount) / quote_fee_amount
+
+                log_output = (
+                    "Quote test:"
+                    "Tx hash: "
+                    + tx_hash
+                    + "\t\t"
+                    + "Order indices: "
+                    + str(i)
+                    + "\t\t"
+                    + "Winning Solver: "
+                    + transaction["from"]
+                    + "\t\t"
+                    + "Fee: "
+                    + str(fee_amount)
+                    + "\t\t"
+                    + "Fee quote: "
+                    + str(quote_fee_amount)
+                    + "\t\t"
+                    + "Absolute difference: "
+                    + str(diff_fee_abs)
+                    + "\t\t"
+                    + "Relative difference: "
+                    + (str(format(100 * diff_fee_rel, ".2f")) + "%")
+                )
+                if abs(diff_fee_rel) > FEE_RELATIVE_DEVIATION_FLAG:
+                    self.alert(log_output)
+
+            batch_cost = gas_amount * gas_price
+
+            # get batch fees (not correct if some orders are market orders)
+            competition_data_list = self.orderbook_api.get_solver_competition_data(
+                [tx_hash]
+            )
+            if len(competition_data_list) == 0:
+                self.logger.debug("No competition data found. Skipping hash.")
+                return False
+            competition_data = competition_data_list[0]
+
+            batch_fee = int(competition_data["solutions"][-1]["objective"]["fees"])
+
+            a_abs = batch_fee - batch_cost
+            a_rel = (batch_fee - batch_cost) / batch_cost
+
+            log_output = (
```

Comment:
Same here. please fix! Also, I am not sure the double tabs are necessary... maybe just one tab.